src/CMakeLists.txt

```diff
@@ -734,6 +734,8 @@ else()
 				SET(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -Wl,-Ttext-segment=0x200000000")
 			endif()
 		endif()
+		# Make symbols that start with "mtffi_" available to LuaJIT's FFI library.
+		SET(CMAKE_EXE_LINKER_FLAGS  "${CMAKE_EXE_LINKER_FLAGS} -Wl,--dynamic-list=${CMAKE_SOURCE_DIR}/src/ffi_dynamic_list.txt")
```

Comment:
Unless this can be solved with `__attribute__((visibility(....)))` (which is still gcc/clang only but at least linker-independent) I think this is good example of why FFI is kinda unportable.

Reply:
It seems `__attribute__((visibility("default")))` doesn't fix the problem, though it might be necessary for Android. It's true that some setup is required, but the FFI interface itself should be portable. In the worst case, FFI could be disabled on incompatible platforms.

Reply:
The argument I'm trying to make is that this is all extra complexity that needs to be supported, question is whether the advantages are worth it.

Reply:
Using FFI, `get_data` is nearly twice as fast by my measurement.

Reply:
The VM methods implemented are very simple. I don't think we should use FFI if it would lead to duplicated complex code.